implementation/src/main/java/io/smallrye/config/ConfigValueConfigSourceInterceptor.java

```diff
@@ -1,23 +1,44 @@
 package io.smallrye.config;
 
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.regex.Pattern;
+
 import javax.annotation.Priority;
 
 import org.jboss.logging.Logger;
 
 @Priority(100)
 public class ConfigValueConfigSourceInterceptor implements ConfigSourceInterceptor {
     private static final Logger LOG = Logger.getLogger("io.smallrye.config");
+    private static final String EXCLUDE_KEYS_PATTERN_KEY = "smallrye.log.exclude.keys.pattern";
+
+    private AtomicBoolean initPattern = new AtomicBoolean(false);
+    private Pattern excludeKeyPattern;
 
     @Override
     public ConfigValue getValue(final ConfigSourceInterceptorContext context, final String name) {
-        final ConfigValue configValue = context.proceed(name);
+        if (!initPattern.get()) {
```

Comment:
This is a poor way to provide atomicity.  If this is called from two threads, one could be initializing still while the exclude pattern is still being built.  The second thread could observe `initPattern` being `true` while `excludeKeyPattern` is still `null`.

Reply:
Ops, you are right.